### PR TITLE
Add extra annotations scores to the filtered variants (#242).

### PR DESCRIPTION
**Pull request by snesic**
This is a clean branch that addresses the previous comments. 

Closes [#242](https://github.com/bihealth/varfish-server/issues/242).

### CODE REVIEW

**Review by holtgrewe — COMMENTED** [review]
Please shorten commit message to ~70 chars.

Otherwise it looks good to me. @stolpeo should also comment.

File: variants/templates/variants/filter_result/header.html [hunk 1]
```diff
@@ -31,6 +31,9 @@
   <th class="effect-change-protein"></th>{# effect change protein #}
   <th class="effect-change-cdna"></th>{# effect change cdna #}
   <th class="distance-to-splicesite"></th>{# distance to splicesite #}
+  {% for value in result_extra_annos_header %}
+    <th class="effect-change-extra_annos-{{ forloop.counter }}"></th>
```
Comment (stolpeo) [inline variants/templates/variants/filter_result/header.html:35]: the class name is weird, this is not an effect-change thing. also, the class can probably be dropped as it not used anyway?

File: variants/templates/variants/filter_result/header.html [hunk 2]
```diff
@@ -97,6 +100,9 @@
   <th class="effect-change-protein">effect protein</th>
   <th class="effect-change-cdna">effect cdna</th>
   <th class="distance-to-splicesite">dist. to splicesite</th>
+  {% for value in result_extra_annos_header %}
+    <th class="effect-change-extra_annos-{{ forloop.counter }}"> {{ value }}</th>
```
Comment (stolpeo) [inline variants/templates/variants/filter_result/header.html:104]: same here

File: variants/templates/variants/filter_result/row.html [hunk 3]
```diff
@@ -387,6 +387,14 @@
   <td class="text-nowrap effect-change-protein">{{ entry.hgvs_p|default:'-' }}</td>
   <td class="text-nowrap effect-change-cdna">{{ entry.hgvs_c|default:'-' }}</td>
   <td class="text-nowrap distance-to-splicesite">{{ entry.exon_dist|default:'-' }}</td>
+  {% if entry.extra_annos is not None %}
+    {% for value in entry.extra_annos.0 %}
+      <td class="text-nowrap effect-change-extra_annos-{{ forloop.counter }}">{{ value|default:'-' }}</td>
+    {% endfor %}
+  {% else %}
+    {% for value in result_extra_annos_header %}
+      <td class="text-nowrap effect-change-extra_annos-{{ forloop.counter }}">{{ '-' }}</td>  {% endfor %}
```
Comment (stolpeo) [inline variants/templates/variants/filter_result/row.html:396]: formatting

File: variants/templates/variants/filter_result/header.html (hunk 1 above)
Comment (stolpeo) [reply variants/templates/variants/filter_result/header.html:35]: ah, seems to be used in the dropdown. naming is still weird.

Comment (stolpeo) [reply variants/templates/variants/filter_result/header.html:35]: https://github.com/bihealth/varfish-server/blob/7538557af9c4fa6fb43f055dc4e75a52947c7e11/varfish/static/js/state_machine.js#L417 this defines which columns have the little asc/desc sort button. please check if this still holds after your addition.

File: variants/templates/variants/filter_result/table.html [hunk 4]
```diff
@@ -210,6 +210,9 @@ <h2>
             <option value="effect-change-protein">Effect Protein</option>
             <option value="effect-change-cdna">Effect cDNA</option>
             <option value="distance-to-splicesite">Distance to SpliceSite</option>
+            {% for value in result_extra_annos_header %}
+              <option value="effect-change-extra_annos-{{ forloop.counter }}"}}>{{ value }}</option>
```
Comment (stolpeo) [inline variants/templates/variants/filter_result/table.html:214]: naming

File: variants/templates/variants/filter_result/row.html [hunk 5]
```diff
@@ -387,6 +387,14 @@
   <td class="text-nowrap effect-change-protein">{{ entry.hgvs_p|default:'-' }}</td>
   <td class="text-nowrap effect-change-cdna">{{ entry.hgvs_c|default:'-' }}</td>
   <td class="text-nowrap distance-to-splicesite">{{ entry.exon_dist|default:'-' }}</td>
+  {% if entry.extra_annos is not None %}
+    {% for value in entry.extra_annos.0 %}
+      <td class="text-nowrap effect-change-extra_annos-{{ forloop.counter }}">{{ value|default:'-' }}</td>
```
Comment (stolpeo) [inline variants/templates/variants/filter_result/row.html:392]: naming

File: variants/templates/variants/filter_result/row.html (hunk 3 above)
Comment (stolpeo) [reply variants/templates/variants/filter_result/row.html:396]: naming

File: variants/tests/test_ui.py [hunk 6]
```diff
@@ -1106,3 +1107,125 @@ def test_variant_joint_filter_download(self):
...
+        remove_from_extra_anno_values = [self.extra_anno_field.label, "-", ""]
+        row_extra_anno_values = [
+            float(i.text)
+            for i in row_extra_anno_items
+            if i.text not in remove_from_extra_anno_values
+        ]
+        # click sort extra anno and see if sorted
+        extra_anno_table_header.click()
+        row_extra_anno_items_sorted = table.find_elements_by_class_name(
+            "effect-change-extra_annos-1"
```
Comment (stolpeo) [inline variants/tests/test_ui.py:1210]: naming

File: variants/tests/test_ui.py [hunk 7]
```diff
@@ -1106,3 +1107,125 @@ def test_variant_joint_filter_download(self):
...
+        table, row_header_items, row_header_values = self._find_table_column_names()
+        self.assertTrue(self.extra_anno_field.label in row_header_values)
+
+        # get extra anno header element
+        extra_anno_table_header = [
+            i for i in row_header_items if i.text == self.extra_anno_field.label
+        ][0]
+
+        # check extra anno values
+        row_extra_anno_items = table.find_elements_by_class_name("effect-change-extra_annos-1")
```
Comment (stolpeo) [inline variants/tests/test_ui.py:1199]: naming

**Review by stolpeo — CHANGES_REQUESTED** [review]
Overall it's looking good, had some remarks. Not sure with the naming of the css class, maybe a copy-paste thing? You should definitely test whether sorting of the table still works.

**Comment by holtgrewe** [comment]
Hi, we want to release the RC next week. If you address all issues until then, we will be happy to include this.

File: variants/templates/variants/filter_result/header.html (hunk 2 above)
Comment (snesic) [reply variants/templates/variants/filter_result/header.html]: effect-change removed

File: variants/templates/variants/filter_result/row.html (hunk 3 above)
Comment (snesic) [reply variants/templates/variants/filter_result/row.html]: effect-change

File: variants/templates/variants/filter_result/table.html (hunk 4 above)
Comment (snesic) [reply variants/templates/variants/filter_result/table.html]: effect-change

File: variants/tests/test_ui.py (hunk 6 above)
Comment (snesic) [reply variants/tests/test_ui.py]: effect-change removed

File: variants/tests/test_ui.py (hunk 7 above)
Comment (snesic) [reply variants/tests/test_ui.py]: effect-change removed

File: variants/templates/variants/filter_result/header.html (hunk 1 above)
Comment (snesic) [reply variants/templates/variants/filter_result/header.html]: class name changed, effect-change removed. The problem might be column 29 which now dynamically changes. Could we allow sort at that position?

Comment (snesic) [reply variants/templates/variants/filter_result/header.html]: Column 29, which is the last one, comes before extra anno columns so none of them will be affected.

File: variants/templates/variants/filter_result/row.html (hunk 3 above)
Comment (snesic) [reply variants/templates/variants/filter_result/row.html]: formatting corrected

**Comment by holtgrewe** [comment]
Important!

Now that #323 has been resolved you will have to amend your commit message by the following three trailing/last lines. Otherwise, the "check required trailers" check that has just been introduced will fail.

```
Closes: #242
Related-Issue: #242
Projected-Results-Impact: none
```

**Review by stolpeo — APPROVED** [review]

**Pull request merged**